File: data/examples/data-class.cpp
class Customer {
private:
    std::string name;
    std::string email;
    std::string address;
    double creditLimit;
    
public:
    // 只有getter和setter，没有其他行为
    std::string getName() const { return name; }
    void setName(const std::string& name) { this->name = name; }
    
    std::string getEmail() const { return email; }
    void setEmail(const std::string& email) { this->email = email; }
    
    std::string getAddress() const { return address; }
    void setAddress(const std::string& address) { this->address = address; }
    
    double getCreditLimit() const { return creditLimit; }
    void setCreditLimit(double limit) { this->creditLimit = limit; }
};

// 行为被放在了其他类中
class CustomerService {
public:
    bool canPurchase(const Customer& customer, double amount) {
        return customer.getCreditLimit() >= amount;
    }
    
    void updateCreditLimit(Customer& customer, double newLimit) {
        customer.setCreditLimit(newLimit);
    }
};

File: data/examples/data-class.java
class Customer {
    private String name;
    private String email;
    private String address;
    private double creditLimit;
    
    // Only getters and setters, no other behavior
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    
    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }
    
    public double getCreditLimit() { return creditLimit; }
    public void setCreditLimit(double creditLimit) { this.creditLimit = creditLimit; }
}

// Behavior is placed in other classes
class CustomerService {
    public boolean canPurchase(Customer customer, double amount) {
        return customer.getCreditLimit() >= amount;
    }
    
    public void updateCreditLimit(Customer customer, double newLimit) {
        customer.setCreditLimit(newLimit);
    }
}
